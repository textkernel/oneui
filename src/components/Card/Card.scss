.container {
    height: 44px;
    padding: var(--spacing-4x) var(--spacing-5x);
    display: flex;
    justify-content: space-between;
    gap: var(--spacing-5x);
    border: {
        radius: var(--border-radius);
        width: var(--line-normal);
        style: solid;
        color: #a6a9b5;
    }
}

.leftColumn {
    display: flex;
    flex-direction: column;
    gap: var(--spacing-2x);
    overflow: hidden;

    .title {
        font-weight: var(--font-weight-normal);
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .subtitle {
        color: #808080;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }
}

.trailing {
    display: flex;
    flex-direction: column;
    justify-content: center;
}
