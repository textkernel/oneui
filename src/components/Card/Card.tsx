import React from 'react';
import { Text } from '../Text';
import { Heading } from '../Heading';
import styles from './Card.scss';

export interface Props {
    title: string;
    subtitle?: string;
    trailing?: string | number;
}

export const Card = ({ title, subtitle, trailing }: Props) => (
    <div className={styles.container}>
        <div className={styles.leftColumn}>
            <Heading level="h3" className={styles.title}>
                {title}
            </Heading>
            {subtitle && (
                <Text inline className={styles.subtitle}>
                    {subtitle}
                </Text>
            )}
        </div>
        {typeof trailing !== 'undefined' && (
            <div className={styles.trailing}>
                <Heading level="h1">{trailing}</Heading>
            </div>
        )}
    </div>
);

Card.displayName = 'Card';
