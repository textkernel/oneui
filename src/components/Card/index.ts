export { Card, Props as CardProps } from './Card';
