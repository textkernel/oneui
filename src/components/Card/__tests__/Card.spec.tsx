import * as React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Card } from '../Card';

describe('Card', () => {
    const title =
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugit expedita labore laborum culpa harum incidunt.';
    const subtitle = 'Fugit expedita labore laborum culpa harum incidunt.';

    it('should render correctly', () => {
        const view = render(<Card title={title} subtitle={subtitle} trailing={1} />);

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('heading', { name: title })).toBeInTheDocument();
        expect(screen.getByText(subtitle)).toBeInTheDocument();
        expect(screen.getByRole('heading', { name: '1' })).toBeInTheDocument();
    });

    it('should render Card only with the title', () => {
        const view = render(<Card title={title} />);

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('heading', { name: title })).toBeInTheDocument();
    });

    it('should render Card only with the title and trailing', () => {
        const view = render(<Card title={title} trailing="1" />);

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('heading', { name: title })).toBeInTheDocument();
        expect(screen.getByRole('heading', { name: '1' })).toBeInTheDocument();
    });
});
