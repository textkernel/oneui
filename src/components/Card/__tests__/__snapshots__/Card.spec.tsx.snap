// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render Card only with the title 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="leftColumn"
    >
      <h3
        class="Heading Heading--level_h3 title"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugit expedita labore laborum culpa harum incidunt.
      </h3>
    </div>
  </div>
</div>
`;

exports[`Card should render Card only with the title and trailing 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="leftColumn"
    >
      <h3
        class="Heading Heading--level_h3 title"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugit expedita labore laborum culpa harum incidunt.
      </h3>
    </div>
    <div
      class="trailing"
    >
      <h1
        class="Heading Heading--level_h1"
      >
        1
      </h1>
    </div>
  </div>
</div>
`;

exports[`Card should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="leftColumn"
    >
      <h3
        class="Heading Heading--level_h3 title"
      >
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Fugit expedita labore laborum culpa harum incidunt.
      </h3>
      <span
        class="subtitle"
      >
        Fugit expedita labore laborum culpa harum incidunt.
      </span>
    </div>
    <div
      class="trailing"
    >
      <h1
        class="Heading Heading--level_h1"
      >
        1
      </h1>
    </div>
  </div>
</div>
`;
